src/Sentry/Internal/ObjectExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// Utility methods to enable storing arbitrary properties on any object, without worrying about garbage collection.
+/// </summary>
+internal static class ObjectExtensions
+{
+    private static ConditionalWeakTable<object, Dictionary<string, object?>> Map { get; } = new();
+
+    private static Dictionary<string, object?> AssociatedProperties(this object source) =>
+        Map.GetValue(source, _ => new Dictionary<string, object?>());
+
+    internal static void SetFused(this object source, string propertyName, object? value) =>
+        AssociatedProperties(source)[propertyName] = value;
+
+    internal static void SetFused<T>(this object source, T value)
```

Comment:
```suggestion
    public static void SetFused<T>(this object source, T value)
```
The setter and getter are supposed to be used, right? And the `internal Fused` is for testing?

Reply:
We could get rid of the `Fused` extension... I'd copied that over [from here](https://github.com/mentaldesk/fuse) (I was looking for an excuse to build/publish a nuget package from scratch, so that I could better learn all the moving parts of Sentry's CI process).

Reply:
That's a super nifty little library!